### PR TITLE
Improved `DatasetNotFound` error message

### PR DESCRIPTION
**Pull request by klakhov**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
  - Enhanced dataset detection and import functionality for various formats including Cityscapes, KITTI, LabelMe, LFW, Market1501, VOC, MOT, OpenImages, Pascal VOC, PointCloud, VGG Face2, WiderFace, and YOLO.

- **Bug Fixes**
  - Improved error handling for dataset import operations, ensuring better detection of incorrect file structures.

- **Tests**
  - Added a new test to verify import errors for incorrect file structures in source archives.

### CODE REVIEW

File: cvat/apps/dataset_manager/bindings.py [hunk 1]
```diff
@@ -1664,6 +1668,33 @@ def GetCVATDataExtractor(
 class CvatImportError(Exception):
     pass
 
+class CvatDatasetNotFoundError(Exception):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1671]: I think it makes sense to inherit this class from `CvatImportError`

File: cvat/apps/dataset_manager/bindings.py [hunk 2]
```diff
@@ -1664,6 +1668,33 @@ def GetCVATDataExtractor(
...
+    message = field(default="")
+    reason = field(default="")
+    format_name = field(default="")
+    _docs_base_url = f"{settings.CVAT_DOCS_URL}/manual/advanced/formats/"
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1675]: If this class is not declared with `@dataclass` or `@attrs.define`, these fields will be class members, not instance fields.

File: cvat/apps/dataset_manager/bindings.py [hunk 3]
```diff
@@ -1664,6 +1668,33 @@ def GetCVATDataExtractor(
 class CvatImportError(Exception):
     pass
 
+class CvatDatasetNotFoundError(CvatImportError):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1671]:
```suggestion
@attrs
class CvatDatasetNotFoundError(CvatImportError):
```

File: cvat/apps/dataset_manager/formats/camvid.py [hunk 4]
```diff
@@ -6,8 +6,7 @@
 from datumaro.components.dataset import Dataset
 from pyunpack import Archive
 
-from cvat.apps.dataset_manager.bindings import (GetCVATDataExtractor,
-    import_dm_annotations)
+from cvat.apps.dataset_manager.bindings import (GetCVATDataExtractor, import_dm_annotations)
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/formats/camvid.py:9]: It seems like this format misses the detection call.

File: cvat/apps/dataset_manager/formats/icdar.py [hunk 5]
```diff
@@ -10,8 +10,7 @@
 from datumaro.components.dataset import Dataset
 from datumaro.components.extractor import ItemTransform
 
-from cvat.apps.dataset_manager.bindings import (GetCVATDataExtractor,
-    import_dm_annotations)
+from cvat.apps.dataset_manager.bindings import (GetCVATDataExtractor, import_dm_annotations)
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/formats/icdar.py:13]: It seems like this format misses the detection call.

File: cvat/apps/dataset_manager/project.py [hunk 6]
```diff
@@ -158,7 +158,7 @@ def import_dataset(self, dataset_file, importer, **options):
         with TemporaryDirectory(dir=temp_dir_base) as temp_dir:
             try:
                 importer(dataset_file, temp_dir, project_data, self.load_dataset_data, **options)
-            except DatasetNotFoundError as not_found:
+            except (DatasetNotFoundError, CvatDatasetNotFoundError) as not_found:
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/project.py:161]:
```suggestion
            except DatasetNotFoundError as not_found:
```

As the class is inherited, no extra handling is needed here.

File: cvat/apps/dataset_manager/bindings.py [hunk 7]
```diff
@@ -2265,3 +2296,19 @@ def load_dataset_data(project_annotation, dataset: dm.Dataset, project_data):
             dataset_files['data_root'] = osp.commonpath(root_paths) + osp.sep
 
         project_annotation.add_task(task_fields, dataset_files, project_data)
+
+def detect_dataset(temp_dir, format_name, importer):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:2300]:
```suggestion
def detect_dataset(dataset_dir: str, format_name: str, importer: Importer) -> None:
```

File: cvat/apps/dataset_manager/bindings.py [hunk 8]
```diff
@@ -1664,6 +1669,33 @@ def GetCVATDataExtractor(
...
+    message: str = ""
+    reason: str = ""
+    format_name: str = ""
+    _docs_base_url = f"{settings.CVAT_DOCS_URL}/manual/advanced/formats/"
+
+    def __str__(self):
+        formatted_format_name = self._format_name_for_docs()
+        docs_message = self._docs_message(formatted_format_name)
+        display_message = self._clean_display_message()
+        return f"{docs_message}. {display_message}"
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1683]: Could you add at least 1 API test for these errors?

File: cvat/apps/dataset_manager/bindings.py [hunk 9]
```diff
@@ -1664,6 +1669,33 @@ def GetCVATDataExtractor(
...
     pass
 
+@attrs
+class CvatDatasetNotFoundError(CvatImportError):
+    message: str = ""
+    reason: str = ""
+    format_name: str = ""
+    _docs_base_url = f"{settings.CVAT_DOCS_URL}/manual/advanced/formats/"
+
+    def __str__(self):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1679]:
```suggestion
    def __str__(self) -> str:
```

File: cvat/apps/dataset_manager/bindings.py [hunk 10]
```diff
@@ -1664,6 +1669,33 @@ def GetCVATDataExtractor(
...
+        formatted_format_name = self._format_name_for_docs()
+        docs_message = self._docs_message(formatted_format_name)
+        display_message = self._clean_display_message()
+        return f"{docs_message}. {display_message}"
+
+    def _format_name_for_docs(self):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1685]:
```suggestion
    def _format_name_for_docs(self) -> str:
```

File: cvat/apps/dataset_manager/bindings.py [hunk 11]
```diff
@@ -1664,6 +1669,33 @@ def GetCVATDataExtractor(
...
+        return self.format_name.replace("_", "-")
+
+    def _docs_message(self, formatted_format_name):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1688]:
```suggestion
    def _docs_message(self, formatted_format_name: str) -> str:
```

File: cvat/apps/dataset_manager/bindings.py [hunk 12]
```diff
@@ -1664,6 +1669,33 @@ def GetCVATDataExtractor(
...
+        return f"Check [format docs]({self._docs_base_url}format-{formatted_format_name})"
+
+    def _clean_display_message(self):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1691]:
```suggestion
    def _clean_display_message(self) -> str:
```

File: tests/python/rest_api/test_tasks.py [hunk 13]
```diff
@@ -2668,6 +2668,27 @@ def test_import_annotations_after_deleting_related_cloud_storage(
         task.import_annotations(self.import_format, file_path)
         self._check_annotations(task_id)
 
+    def test_check_import_error_on_wrong_file_structure(self, tasks_with_shapes):
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_tasks.py:2671]: LGTM, but as this PR affects different formats, it makes sense to run this test for other formats as well. Consider adding `@pytest.mark.parametrize(format_name, ["...", ...])` with the list of formats.

File: tests/python/rest_api/test_tasks.py [hunk 14]
```diff
@@ -2668,6 +2668,34 @@ def test_import_annotations_after_deleting_related_cloud_storage(
         task.import_annotations(self.import_format, file_path)
         self._check_annotations(task_id)
 
+    @pytest.mark.parametrize(
+        "format_name", ["COCO 1.0", "COCO Keypoints 1.0", "CVAT 1.1", "LabelMe 3.0",
+            "MOT 1.1", "MOTS PNG 1.0", "PASCAL VOC 1.1", "Segmentation mask 1.1",
+            "YOLO 1.1", "WiderFace 1.0", "VGGFace2 1.0", "Market-1501 1.0",
+            "Kitti Raw Format 1.0", "Sly Point Cloud Format 1.0", "KITTI 1.0",
+            "LFW 1.0", "Cityscapes 1.0", "Open Images V6 1.0", "Datumaro 1.0",
+            "Datumaro 3D 1.0"])
```
[the remark this answers is not in the document]
Comment (klakhov) [reply tests/python/rest_api/test_tasks.py]: This test tests exactly this error. We check the error message contents for different formats

**Pull request merged**